tests/spread/store/charm-upload-and-release/task.yaml

```diff
@@ -42,10 +42,20 @@ execute: |
   # release that last revision to edge
   charmcraft release $CHARM_DEFAULT_NAME -r $uploaded_revno -c edge
 
-  # validate the channel map
-  edge_release=$(charmcraft status $CHARM_DEFAULT_NAME --format=json | jq -r '.[] | select(.track=="latest") | .mappings[0].releases | .[] | select(.channel=="latest/edge")')
-  edge_revision=$(echo $edge_release | jq -r .revision)
-  # check that the current release greater or equal than what was
-  # uploaded, because other tests running concurrently may have
-  # also released (but for sure cannot be a previous revision)
-  test $edge_revision -ge $uploaded_revno
+  # Releases may have some delay in the craft store, so sleep for a bit before trying and try up to 10 times.
+  for i in {1..10}
+  do
```

Comment:
should we have a max number of tries here to keep this from looping forever?

Reply:
This only tries up to 10 times.

Reply:
![image](https://github.com/canonical/charmcraft/assets/1951299/86ddc891-0713-48e2-90f2-2b8435011f88)

Reply:
![](https://media3.giphy.com/media/3o6fJbwYFe3SmVVQ4M/giphy.gif)